### PR TITLE
Task15 16

### PR DESCRIPTION
**Pull request by torniko111**
1. Task15. Added RabbitMq with Hangfire so its sending Mails Using free Fake Smtp Ethereal.mail. 

2.  Task16. Added Flag to Appsetting. that controls sending mail trought rabbit, or directly.  If task16 meaning that i have to create local smtp server to send mail, it needs private third party email hosting. As i testad it with hmail. so i think its just meaning as i did.
3. removed call api from unit tests, so created dictionary of Celcius and Cities for testing purposes. so tests are independent now.
4.  added reports unit tests, created mock list of weather objects.

at the end i will refactor program.cs and relocate hangfires methods from here to another class.

### CODE REVIEW

File: WeatherForecast/BL/Interfaces/IWeatherService.cs [hunk 1]
```diff
@@ -9,7 +10,7 @@ namespace BL.Interfaces
 {
     public interface IWeatherService
     {
-        Task<Weather> AddAsync(Weather user);
+        void AddAsync(Weather user);
```
Comment (svadrif) [inline WeatherForecast/BL/Interfaces/IWeatherService.cs]: if this method is not async, then you should delete Async postfix

File: WeatherForecast/BL/Interfaces/IWeatherService.cs [hunk 2]
```diff
@@ -19,5 +20,8 @@ public interface IWeatherService
         Task GetCurrentWeatherByCitiesSameTime(string[] city);
         Task<List<Weather>> Getreport(DateTime from, DateTime to, string city);
         Task<Dictionary<double, string>> GetWeatherForecast(string city, int days);
+        Task<string> AverageStatistics(string city, string period);
```
Comment (svadrif) [inline WeatherForecast/BL/Interfaces/IWeatherService.cs:22]: async methods should end with Async postfix

File: WeatherForecast/BL/Interfaces/IWeatherService.cs (hunk 1 above)
Comment (torniko111) [reply WeatherForecast/BL/Interfaces/IWeatherService.cs]: was not nessesery at all, and removed.        void method

File: WeatherForecast/BL/Interfaces/IWeatherService.cs (hunk 2 above)
Comment (torniko111) [reply WeatherForecast/BL/Interfaces/IWeatherService.cs:22]: updated with async await keyword.

**Review by svadrif — APPROVED** [review]

**Pull request merged**